app/models/Missions.js

```diff
@@ -73,10 +74,38 @@ var MissionSchema = function(outcomeType) {
 util.inherits(MissionSchema, Schema);
 
 var missionSchema = new MissionSchema();
-missionSchema.pre('save', function(next) {
+missionSchema.pre('save', function(next){
     var that = this;
     var missionType = that.getType();
 
+    if(that.isFirstOfType === undefined){
+        that.getMissionCount(missionType, that.location, function(err, count){
+            if(err){
+                return next(err);
+            }
+            if (count <= 0){
+                that.isFirstOfType = true;
+            }else{
+                that.isFirstOfType = false;
+            }
+            return next();
+        });
+    }
```

Comment:
shouldn't there be a:
```
else {
    return next();
}
```
Otherwise it will never continue if isFirstOfType is already defined.